neural_compressor/adaptor/onnxrt.py

```diff
@@ -579,6 +598,85 @@ def _requantize_bias(self, model, bias_name, bias_data):
+        initializer_names = [init.name for init in model.model.graph.initializer]
+        pattern = ".*word.*embedding.*"
+        for name in initializer_names:
+            obj = re.findall(pattern, name)
+            if len(obj) > 0:
+                is_nlp = True
+                break
+        
+        # 2. according to input
+        # typically, NLP models have multiple inputs, 
+        # and the dimension of each input is usually 2 (batch_size, max_seq_len)
+        sess = ort.InferenceSession(model.model.SerializeToString())
+        input_shape_lens = [len(input.shape) for input in  sess.get_inputs()]
+        if len(input_shape_lens) > 1 and all(shape_len == 2 for shape_len in input_shape_lens):
+            is_nlp = True
+
+        # 3. according to attention structure
+        for node in model.model.graph.node:
+            if node.op_type == 'Add':
+                start_node = node
+                qkv_nodes_list = [
+                    # match base attention structure
+                    model.match_parent_path(
+                        start_node,
+                        ["Add", "MatMul", "Reshape", "Transpose", "MatMul"],
+                        [0, None, 0, 0, 0],),
+                    model.match_parent_path(
+                        start_node,
+                        ["Add", "MatMul", "Reshape", "Transpose", "MatMul"],
+                        [1, None, 0, 0, 0]),
+
+                    # match gpt attention no past structure
+                    model.match_parent_path(
+                        start_node,
+                        ["Reshape", "Gemm", "Reshape", "Reshape", "Transpose", "MatMul"],
+                        [ None, 0, 0, 0, 0, 0],
+                        output_name_to_node=model.output_name_to_node,
+                        return_indice=[])
+                    ]
+                if not any(qkv_nodes_list):
+                    continue
+                qkv_nodes = [qkv for qkv in qkv_nodes_list if qkv is not None][-1]
+                other_inputs = []
+                for input in start_node.input:
+                    if input not in model.output_name_to_node:
+                        continue
+                    if input == qkv_nodes[0].output[0]:
+                        continue
+                    other_inputs.append(input)
+                if len(other_inputs) != 1:
+                    continue
+                root_input = other_inputs[0]
+                input_name_to_nodes = model.input_name_to_nodes
+                children = input_name_to_nodes[root_input]
+                children_types = [child.op_type for child in children]
+                if children_types.count("MatMul") == 3:
+                    is_nlp = True
+                    break
+
+        # 4. according to LSTM structure
+        if "LSTM" in [node.op_type for node in model.model.graph.node]:
+            is_nlp = True
+
+        logger.warning("The model is automatically detected as {}belonging to NLP domain. "
```

Comment:
"sth detected as belonging to NLP domain"

the english syntax looks not correct